src/patchelf.cc

```diff
@@ -1745,21 +1754,81 @@ void ElfFile<ElfFileParamNames>::addNeeded(const std::set<std::string> & libs)
     changed = true;
 }
 
+// https://stackoverflow.com/a/478960/143733
+std::string exec(const char* cmd) {
+    std::array<char, 128> buffer;
+    std::string result;
+    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
```

Comment:
Ideally this should also check errors returned from `ld` i.e. if one tries to patchelf an invalid file.

Reply:
Added a check but I couldn't express it since there are tons of checks beforehand that validate that the file given is an ELF file already.